### PR TITLE
Add clear()

### PR DESCRIPTION
**Pull request by vweevers**
Start of #236. For early feedback (see comments in code).

### CODE REVIEW

File: abstract-leveldown.js [hunk 1]
```diff
@@ -179,6 +179,62 @@ AbstractLevelDOWN.prototype._batch = function (array, options, callback) {
...
+
+  // Because the returned records are the same, reverse only makes sense in
+  // combination with limit (e.g. to remove the last N records).
+  options.reverse = !!options.reverse
+  options.limit = 'limit' in options ? options.limit : -1
+
+  this._clear(options, callback)
+}
+
+AbstractLevelDOWN.prototype._clear = function (options, callback) {
```
Comment (vweevers) [inline abstract-leveldown.js]: Given the issues described below, perhaps the default `_clear` should just be a noop.

Comment (ralphtheninja) [reply abstract-leveldown.js]: Makes sense. Not sure which one though.

Comment (vweevers) [reply abstract-leveldown.js]: There's no precedent for throwing, but a noop is undesirable.. unless we enforce the presence of `clear` through tests.

I guess the question is: do we aim to implement `clear` everywhere? Or do we start with partial support, in which case throwing makes more sense.

Comment (vweevers) [reply abstract-leveldown.js]: Actually, I prefer putting this feature on hold. V6 already has lots of breaking changes. We have a release dance ahead.

**Comment by vweevers** [comment]
@ralphtheninja Updated. Could you have a look? Tests are yet to do, but I'd first like a review of the general approach, taking into account https://github.com/Level/memdown/pull/191 and https://github.com/Level/level-js/issues/174.

File: test/common.js [hunk 2]
```diff
@@ -19,7 +31,16 @@ function testCommon (options) {
...
     snapshots: options.snapshots !== false,
-    seek: options.seek !== false
+    seek: options.seek !== false,
+    clear: clear
+  }
+}
+
+function warn (msg) {
+  if (typeof process !== 'undefined' && process && process.emitWarning) {
+    process.emitWarning(msg)
```
Comment (ralphtheninja) [inline test/common.js]: TIL!

**Comment by vweevers** [comment]
Done. Tested against https://github.com/Level/memdown/pull/191.

**Comment by vweevers** [comment]
Once https://github.com/Level/memdown/pull/191 lands in master I'll make a canary branch there to run browser tests as well.

**Pull request merged**